src/screens/Home/ResourcesItemScreen.js

```diff
@@ -78,37 +103,102 @@ export default function ResourcesItemScreen(props) {
       setType(resourceData.type);
       setSubheader(resourceData.subheader);
 
-      /* Different behavior if resource is of type STEPS or FREEFORM */
-      if (type === 'STEPS') {
-        firestore().collection('resource_items').doc(resourceID).collection('steps')
-          .get()
-          .then((stepsList) => {
-            const stepsData = stepsList.docs.map((doc) => ({ ...doc.data(), id: doc.id }));
+      /* Query all map markers */
+      firestore().collection('resource_items').doc(resourceID).collection('mapMarkers')
+        .get()
+        .then((markersList) => {
+          const markersData = markersList.docs.map((doc) => ({ ...doc.data(), id: doc.id }));
 
-            const sortedStepsData = stepsData.sort((a, b) => {
-              if (a.stepNumber > b.stepNumber) {
-                return 1;
-              }
-              if (a.stepNumber < b.stepNumber) {
-                return -1;
+          if (markersData.length) {
+            /* Calculates the average latitude and longitude to give the center of all markers */
+            const avgCoord = markersData.reduce((total, amount, index, array) => {
+              const result = {
+                latitude: total.latitude + amount.latitude,
+                longitude: total.longitude + amount.longitude,
+              };
+
+              if (index === array.length - 1) {
+                return {
+                  latitude: result.latitude / (array.length),
+                  longitude: result.longitude / (array.length),
+                };
               }
-              return 0;
+              return result;
             });
 
-            setSteps(sortedStepsData.map((step) => (
-              <ResourceStep
-                stepNumber={step.stepNumber}
-                title={step.title}
-                description={step.body}
-                key={step.id}
+            /* Calculates maximum distance between center and a marker to size map deltas */
+            let latAccum = 0.0;
+            let lonAccum = 0.0;
+            setMaxDelta(markersData.reduce((_, amount, index, array) => {
```

Comment:
Is there anyway to do this without using `let`? I prefer not to have side effects like this, but I don't know how that would work.

If not, you can use `forEach` instead of `reduce` to find the max lats and longs.